hack/netpol/pkg/utils/k8s_util.go

```diff
@@ -0,0 +1,276 @@
+package utils
+
+import (
+
+	//	"context"
+	"bytes"
+	"fmt"
+	"k8s.io/client-go/rest"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	v1net "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/remotecommand"
+)
+
+type Kubernetes struct {
+	podCache  map[string][]v1.Pod
+	ClientSet *kubernetes.Clientset
+}
+
+func NewKubernetes() (*Kubernetes, error) {
+	clientSet, err := Client()
+	if err != nil {
+		return nil, errors.WithMessagef(err, "unable to instantiate kube client")
+	}
+	return &Kubernetes{
+		podCache:  map[string][]v1.Pod{},
+		ClientSet: clientSet,
+	}, nil
+}
+
+func (k *Kubernetes) GetPods(ns string, key, val string) ([]v1.Pod, error) {
+	if p, ok := k.podCache[fmt.Sprintf("%v_%v_%v", ns, key, val)]; ok {
+		return p, nil
+	}
+
+	v1PodList, err := k.ClientSet.CoreV1().Pods(ns).List(metav1.ListOptions{})
```

Comment:
i think you can rely on client-go informers and perhaps generate a labels.Selector based on your labels and pass it to List() and then not have to check for every pod and their labels yourself 
eg
metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
selector := metav1.LabelSelector{MatchLabels: map[string]string{key: value}}
pLabelSelector, _ := metav1.LabelSelectorAsSelector(&selector)
and then pass pLabelSelector to the podLister
you can use podInformer.Lister() from "k8s.io/client-go/informers/core/v1"

Reply:
yeah, i can use a `Selector`, will push this change .